Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/AbfsUri.java

```diff
@@ -57,42 +57,43 @@ public String getAccountName() {
 
     @Override
     public String getPath() {
-        return relativePath.get();
+        return URI.create(path.get()).getPath();
     }
 
     @Override
-    AzureStorageUri resolve(String path) {
-        return AbfsUri.parse(getUri().resolve(path).toString());
+    public String getRawPath() {
+        return path.get();
     }
 
     @Override
-    AzureStorageUri normalizeWithSlashEnding() {
-        return AbfsUri.parse(UriUtil.normalizeWithSlashEnding(getUri()).toString());
+    AzureStorageUri parseUri(String encodedUri) {
```

Comment:
The type might have to be aligned with name.

```suggestion
    AzureStorageUri parseUri(URI encodedUri) {
```